### ISSUE TITLE
COM818 with ruff format fixes the error too harshly

### ISSUE THREAD

**Issue by Neikow** [astral-sh/ruff#14427]
Hi ! Thanks for providing us with such a great tool !

When the rule COM818 is turned on and we run the formatter, it will wrap the trailing comma with paranthesis so when we run the linter again, the error will be corrected although it was probably a mistake to begin with.

[https://play.ruff.rs/3d855131-5721-4bae-9a93-fb5522dd5637](Playground URL)

I think it would be better to prevent the formatter to run on these types of issues when this particular rule is selected as it can lead to hard to spot errors.

- Commands: ruff format / ruff lint
- Version: v0.7.4
- Relevant settings:
```
{
  "lint":
    "select": [
       "COM818"
    ]
}
```

**Comment by MichaReiser** [comment]
As note for myself:

The input is

```py
test = {"ere": ""},
```

which gets formatted to

```py
test = ({"ere": ""},)
```
Which is no longer flagged by `COM818`. 

Can you tell me more about your workflow? Our general recommendation is to run the linter (and apply fixes) first, then run the formatter.  That the formatter wraps bare tuples in parentheses was also a deliberate decision to make them more visible in, e.g., the editor's case. It also helped to identify many unwanted bare tuples in existing code bases, e.g., Django. But I don't think we ever received feedback on this. So thanks for reporting this specific case where I can see that it can lead to missed possibly-bare tuples.

**Comment by tylerlaprade** [comment]
I have the same problem. When I `⌘ s` on a file containing `x = 2,` in VSCode, parentheses are incorrectly added around `2,`, which is not what I want. The correct code is `x=2`. This happens even if I add `COM818` to `unsafe-fixes` - that seems it's due to the formatter above.
